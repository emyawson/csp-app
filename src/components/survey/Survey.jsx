import React, { Fragment } from 'react';
import { ReactComponent as Angry } from '../../../public/assets/img/smiley_angry_gray.svg';
import { ReactComponent as Annoyed } from '../../../public/assets/img/smiley_annoyed_gray.svg';
import { ReactComponent as Happy } from '../../../public/assets/img/smiley_happy_gray.svg';
import { ReactComponent as Neutral } from '../../../public/assets/img/smiley_neutral_gray.svg';
import { ReactComponent as VeryHappy } from '../../../public/assets/img/smiley_very_happy_gray.svg';

// styles
import './Survey.scss';

const Survey = () => (
        <Fragment>
            <div className="container">
                <div className="container__title">
                    <h4>How was your experience today?</h4>
                </div>
                <div className="container__images">
                    <Angry alt="Angry" className="container__images__angry" />
                    <Annoyed alt="Annoyed" className="container__images__annoyed" />
                    <Happy alt="Neutral" className="container__images__neutral" />
                    <Neutral alt="Happy" className="container__images__happy" />
                    <VeryHappy alt="Very Happy" className="container__images__very-happy" />
                </div>
            </div>
        </Fragment>
);

export default Survey;
