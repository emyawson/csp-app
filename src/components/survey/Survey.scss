.container {
  img {
    cursor: pointer;
  }

  img:active {
    width: 27px;
  }

  &__images {
    display: grid;
    gap: 20px;
    grid-template-columns: repeat(6, 28px);
    height: auto;
    margin-bottom: 15px;
    margin-top: 10px;
    width: 28px;

    &__angry:hover {
      background: url('../../../public/assets/img/smiley_angry.svg') no-repeat;
    }

    &__annoyed:hover {
      background: url('../../../public/assets/img/smiley_annoyed.svg') no-repeat; 
    }

    &__neutral:hover {
      background: url('../../../public/assets/img/smiley_neutral.svg') no-repeat; 
    }

    &__happy:hover {
      background: url('../../../public/assets/img/smiley_happy.svg') no-repeat; 
    }

    &__very-happy:hover {
      background: url('../../../public/assets/img/smiley_very_happy.svg') no-repeat; 
    }
  }
};
