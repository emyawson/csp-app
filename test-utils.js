import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
