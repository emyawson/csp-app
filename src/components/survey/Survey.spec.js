import React from 'react';
// import { render, screen } from "@testing-library/react";
import Survey from './Survey.jsx';
// import { renderWithRouter } from "../test-utils";
import { renderWithRouter } from '../../../test-utils';

const defaultState = {};

const renderComponent = (state = {}) => {
  const combinedState = {
    ...defaultState,
    ...state,
  };
  return renderWithRouter(<Survey />, combinedState);
};

test('renders <Survey/>', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();

  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
