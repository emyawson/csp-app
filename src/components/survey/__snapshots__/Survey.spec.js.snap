// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Survey/> 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container__title"
    >
      <h4>
        How was your experience today?
      </h4>
    </div>
    <div
      class="container__images"
    >
      <reactcomponent
        alt="Angry"
        class="container__images__angry"
      />
      <reactcomponent
        alt="Annoyed"
        class="container__images__annoyed"
      />
      <reactcomponent
        alt="Neutral"
        class="container__images__neutral"
      />
      <reactcomponent
        alt="Happy"
        class="container__images__happy"
      />
      <reactcomponent
        alt="Very Happy"
        class="container__images__very-happy"
      />
    </div>
  </div>
</div>
`;
